src/electron/preview.tsx

```diff
@@ -0,0 +1,321 @@
+			<React.Fragment>
+				<PreviewComponent
+					contents={component.contents}
+					pattern={component.pattern}
+					properties={component.properties}
+					name={component.name}
+					uuid={component.uuid}
+				/>
+				<PreviewHighlight />
+			</React.Fragment>
+		);
+	}
+}
+
+interface PreviewComponentProps {
+	contents: {
+		[slot: string]: PreviewComponentProps[];
+	};
+	name: string;
+	pattern: string;
+	// tslint:disable-next-line:no-any
+	properties: { [key: string]: any };
+	uuid: string;
+}
+
+interface InjectedPreviewComponentProps extends PreviewComponentProps {
+	highlight: HighlightArea;
+	// tslint:disable-next-line:no-any
+	store: any;
+}
+
+@MobXReact.inject('highlight', 'store')
+@MobXReact.observer
+class PreviewComponent extends React.Component<PreviewComponentProps> {
+	public componentWillUpdate(): void {
+		const props = this.props as InjectedPreviewComponentProps;
+
+		if (props.uuid === props.store.get('elementId')) {
+			const node = ReactDom.findDOMNode(this);
+			if (node) {
+				props.highlight.show(node as Element, props.uuid);
+				setTimeout(() => {
+					props.store.set('elementId', null);
+				}, 500);
+			}
+		}
+	}
+
+	public render(): JSX.Element | null {
+		const props = this.props as InjectedPreviewComponentProps;
+		const contents = props.contents ? props.contents : {};
+		const children = typeof contents.default === 'undefined' ? [] : contents.default;
+		const slots = omit(contents, ['default']);
+
+		// Access elementId in render method to trigger MobX subscription
+		props.store.get('elementId');
+
+		const Component = getComponent(props);
+
+		if (!Component) {
+			return null;
+		}
+
+		return (
+			<Component {...slots} {...props.properties} data-sketch-name={props.name}>
+				{children.map(child => <PreviewComponent key={child.uuid} {...child} />)}
+			</Component>
+		);
+	}
+}
+
+interface InjectedPreviewHighlightProps {
+	highlight: HighlightArea;
+}
+
+@MobXReact.inject('highlight')
+@MobXReact.observer
+class PreviewHighlight extends React.Component {
+	public render(): JSX.Element {
+		const props = this.props as InjectedPreviewHighlightProps;
+		const { highlight } = props;
+		const p = highlight.getProps();
+
+		return (
+			<div
+				style={{
+					position: 'absolute',
+					boxSizing: 'border-box',
+					border: '1px dashed rgba(55, 55, 55, .5)',
+					background: `
+					repeating-linear-gradient(
+						135deg,
+						transparent,
+						transparent 2.5px,rgba(51, 141, 222, .5) 2.5px,
+						rgba(51,141,222, .5) 5px),
+						rgba(102,169,230, .5)`,
+					transition: 'all .25s ease-in-out',
+					bottom: p.bottom,
+					height: p.height,
+					left: p.left,
+					opacity: p.opacity,
+					right: p.right,
+					top: p.top,
+					width: p.width
+				}}
+			/>
+		);
+	}
+}
+
+interface TreeNode {
+	contents: {
+		[slot: string]: PreviewComponentProps[];
+	};
+	pattern: string;
+}
+
+// tslint:disable-next-line:no-any
+function getComponent(props: any): string | React.SFC<any> | null {
```

Comment:
It would be great if we could type this. We assume that these props have a pattern and a name property.
But if we look at the place where the function is called the thrown in props are typed as `InjectedPreviewComponentProps` which does not have a pattern or a name property

Reply:
Maybe just type it as a Partial in cases like these. Still better than any :)